src/uct/sm/cma/cma_md.c

```diff
@@ -59,7 +59,8 @@ static ucs_status_t uct_cma_md_open(const char *md_name, const uct_md_config_t *
         .mem_free     = (void*)ucs_empty_function_return_success,
         .mkey_pack    = (void*)ucs_empty_function_return_success,
         .mem_reg      = uct_cma_mem_reg,
-        .mem_dereg    = (void*)ucs_empty_function_return_success
+        .mem_dereg    = (void*)ucs_empty_function_return_success,
+        .mem_type_detect   = ucs_empty_function_return_unsupported,
```

Comment:
why we don't return UCT_MD_MEM_TYPE_DEFAULT for all the rest non-cuda devices ?

Reply:
ucp function will call mem detect only if md supports non default mem type.  I think we can't simply return default for any address without detection logic.

Reply:
so, you want to have the branch on the UCP level, right ?

Reply:
yes

Reply:
hmmm ... how ? You need some sort of capabilities flag for this.

Reply:
this PR also adding mem  type in  md_attr (md_attr->cap.mem_type). So, will check this flag.
UCP changes are here: https://github.com/openucx/ucx/pull/1923

Reply:
okay.